docs/docs/workflows/terraform-module.md

```diff
@@ -0,0 +1,84 @@
+# Terraform module
+It's possible to manage Constellation, including upgrades, through Terraform.
```

Comment:
How does this workflow relate to other uses of Terraform (or self-managed) documented elsewhere in our docs? E.g., does it replace them? Or do we need cross-references?

Reply:
talked to @msanft and agreed to merge the `self-managed` and `terraform` tabs. We also want to refer to the Terraform module as recommended way for self-managed